utils/ssh.py

```diff
@@ -197,20 +201,56 @@ def get_transport(self, *args, **kwargs):
             self.connect()
         return super(SSHClient, self).get_transport(*args, **kwargs)
 
-    def run_command(self, command, timeout=RUNCMD_TIMEOUT, reraise=False, ensure_host=False):
+    def run_command(
+            self, command, timeout=RUNCMD_TIMEOUT, reraise=False, ensure_host=False,
+            ensure_user=False):
+        """Run a command over SSH.
+
+        Args:
+            command: The command. Supports taking dicts as version picking.
+            timeout: Timeout after which the command execution fails.
+            reraise: Does not muffle the paramiko exceptions in the log.
+            ensure_host: Ensure that the command is run on the machine with the IP given, not any
+                container or such that we might be using by default.
+            ensure_user: Ensure that the command is run as the user we logged in, so in case we are
+                not root, setting this to True will prevent from running sudo.
+
+        Returns:
+            A :py:class:`SSHResult` instance.
+        """
         if isinstance(command, dict):
             command = version.pick(command)
         logger.info("Running command `%s`", command)
+        command_changed = False
+        uses_sudo = False
         if self.is_container and not ensure_host:
             command = 'docker exec {} bash -c {}'.format(self._container, quote(
                 'source /etc/default/evm; ' + command))
+            command_changed = True
+
+        if self.username != 'root' and not ensure_user:
+            # We need sudo
+            if ';' in command:
```

Comment:
There's some potential here for an escaped ';' in a command to break this parsing.